src/com/github/igotyou/FactoryMod/Factorys/Compactor.java

```diff
@@ -0,0 +1,303 @@
+        return cp.getEnergyTime();
+    }
+    
+    public double getProductionTime() {
+        return cp.getProductionTime();
+    }
+    
+    public ItemList<NamedItemStack> getInputs() {
+        ItemList<NamedItemStack> inputs = new ItemList<NamedItemStack>();
+        Inventory inv = getInventory();
+        if(mode == CompactorMode.DECOMPACT) {
+            for(ItemStack is : inv.getContents()) {
+                if(is.getItemMeta().hasLore() && is.getItemMeta().getLore().contains(cp.getCompactLore())) {
+                    inputs.add(new NamedItemStack(is.getType(), is.getMaxStackSize(), is.getDurability(), is.getItemMeta().getDisplayName()));
+                    return inputs;
+                }
+            }
+            inputs.add(new NamedItemStack(Material.AIR, 1, (short)1, "Compacted Item"));
+        } else if (mode == CompactorMode.COMPACT) {
+            for(ItemStack is : inv.getContents()) {
+                if(is.getAmount() == is.getMaxStackSize()) {
+                    if(!is.getItemMeta().hasLore()) {
+                        inputs.add(new NamedItemStack(is.getType(), is.getAmount(), is.getDurability(), is.getItemMeta().getDisplayName()));
+                        inputs.addAll(cp.getRecipeMaterials());
+                        return inputs;
+                    }
+                }
+            }
+            return cp.getRecipeMaterials();
+        }
+        return inputs;
+    }
+    
+    public ItemList<NamedItemStack> getOutputs() {
+        ItemList<NamedItemStack> outputs = new ItemList<NamedItemStack>();
+        Inventory inv = getInventory();
+        if(mode == CompactorMode.DECOMPACT) {
+            for(ItemStack is : inv.getContents()) {
+                if(is.getItemMeta().hasLore() && is.getItemMeta().getLore().contains(cp.getCompactLore())) {
+                    outputs.add(new NamedItemStack(is.getType(), is.getMaxStackSize(), is.getDurability(), is.getItemMeta().getDisplayName()));
+                    return outputs;
+                }
+            }
+            outputs.add(new NamedItemStack(Material.AIR, 64, (short)1, "Decompacted Items"));
+        } else if (mode == CompactorMode.COMPACT) {
+            for(ItemStack is : inv.getContents()) {
+                if(is.getAmount() == is.getMaxStackSize()) {
+                    NamedItemStack compactedItemStack = new NamedItemStack(is.getType(), 1, is.getDurability(), is.getItemMeta().getDisplayName());
+                    compactedItemStack.getItemMeta().getLore().add(cp.getCompactLore());
+                    outputs.add(compactedItemStack);
+                    return outputs;
+                }
+            }
+            outputs.add(new NamedItemStack(Material.AIR, 1, (short)1, "Compacted Item"));
+        }
+        return outputs;
+    }
+    
+    public ItemList<NamedItemStack> getRepairs() {
+        return cp.getRepairMaterials();
+    }
+    
+    public void update() {
+        if(mode == CompactorMode.REPAIR) {
```

Comment:
https://github.com/Bergecraft/FactoryMod/commits/iorefactor

Jun 1 and forward.

I am going to merge tealnerd to bergecraft master so that I can merge it into iorefactor, then I can make a clean pull request.